### PR TITLE
Update e2e framework for ILB - Part 2

### PR DESCRIPTION
**Pull request by spencerhance**
Add support for regional resources to gcp.go with minimal changes to the existing fuzzer logic
This allows for whitebox and GC testing of l7-ilb

### CODE REVIEW

File: pkg/fuzz/gcp.go [hunk 1]
```diff
@@ -268,9 +291,15 @@ func hasBetaResource(resourceType string, validators []FeatureValidator) bool {
 
 // GCLBForVIP retrieves all of the resources associated with the GCLB for a
 // given VIP.
-func GCLBForVIP(ctx context.Context, c cloud.Cloud, vip string, validators []FeatureValidator) (*GCLB, error) {
+func GCLBForVIP(ctx context.Context, c cloud.Cloud, vip string, region string, validators []FeatureValidator) (*GCLB, error) {
```
Comment (bowei) [inline pkg/fuzz/gcp.go]: let's make the params a struct
```
type GCLBForVIPParams {
  VIP string
  Region string
  Validators []FeatureValidator
}

func GCLBForVIP(ctx context.Context, c cloud.Cloud, params *GCLBForVIPParams) (...) {
  ...
```

File: pkg/fuzz/gcp.go [hunk 2]
```diff
@@ -491,6 +525,243 @@ func GCLBForVIP(ctx context.Context, c cloud.Cloud, vip string, validators []Fea
 	return gclb, err
 }
 
+// GCLBForVIP retrieves all of the resources associated with the GCLB for a
```
Comment (bowei) [inline pkg/fuzz/gcp.go]: lint error

File: pkg/fuzz/gcp.go [hunk 3]
```diff
@@ -491,6 +525,243 @@ func GCLBForVIP(ctx context.Context, c cloud.Cloud, vip string, validators []Fea
...
+// given VIP.
+func RegionalGCLBForVIP(ctx context.Context, c cloud.Cloud, gclb *GCLB, vip string, region string, validators []FeatureValidator) error {
+
```
Comment (bowei) [inline pkg/fuzz/gcp.go]: delete \n

File: pkg/fuzz/gcp.go [hunk 4]
```diff
@@ -491,6 +525,243 @@ func GCLBForVIP(ctx context.Context, c cloud.Cloud, vip string, validators []Fea
...
+		if hasBetaResource("backendService", validators) {
+			bs, err := c.BetaRegionBackendServices().Get(ctx, bsKey)
+			if err != nil {
+				return err
+			}
+			gclb.BackendService[*bsKey].Beta = bs
+		}
+	}
+
+	negKeys := []*meta.Key{}
```
Comment (bowei) [inline pkg/fuzz/gcp.go]: var negKeys []*meta.Key
igKeys // same

File: pkg/fuzz/gcp.go [hunk 5]
```diff
@@ -491,6 +525,243 @@ func GCLBForVIP(ctx context.Context, c cloud.Cloud, vip string, validators []Fea
...
+	igKeys := []*meta.Key{}
+	// Fetch NEG Backends
+	for _, bsKey := range bsKeys {
+		beGroups := []string{}
```
Comment (bowei) [inline pkg/fuzz/gcp.go]: var beGroups []string

**Comment by spencerhance** [comment]
@bowei Can you take another look?

File: cmd/e2e-test/basic_test.go [hunk 6]
```diff
@@ -187,7 +191,7 @@ func TestEdge(t *testing.T) {
 			t.Logf("GCLB resources createdd (%s/%s)", s.Namespace, tc.ing.Name)
 
 			// Perform whitebox testing.
-			gclb := whiteboxTest(ing, s, t)
+			gclb := whiteboxTest(ing, s, t, "")
```
Comment (bowei) [inline cmd/e2e-test/basic_test.go]: fix in follow up, for literals like empty string "", add comment in front `/*paramName*/""` to aid the reader

**Pull request merged**